### PR TITLE
Update CHANGELOG.md for 1.15.1, 1.14.5, 1.13.9

### PR DESCRIPTION
**Pull request by cinthiaconti**
Update CHANGELOG.md for 1.15.1, 1.14.5, 1.13.9

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,62 @@
...
+IMPROVEMENTS:
+
+* api/plugins: add `tls-server-name` arg for plugin registration [[GH-23549](https://github.com/hashicorp/vault/pull/23549)]
+* auto-auth/azure: Support setting the `authenticate_from_environment` variable to "true" and "false" string literals, too. [[GH-22996](https://github.com/hashicorp/vault/pull/22996)]
+* secrets-sync (enterprise): Added telemetry on number of destinations and associations per type.
+* ui: Adds a warning when whitespace is detected in a key of a KV secret [[GH-23702](https://github.com/hashicorp/vault/pull/23702)]
+* ui: Adds toggle to KV secrets engine value download modal to optionally stringify value in downloaded file [[GH-23747](https://github.com/hashicorp/vault/pull/23747)]
+* ui: Surface warning banner if UI has stopped auto-refreshing token [[GH-23143](https://github.com/hashicorp/vault/pull/23143)]
+* ui: show banner when resultant-acl check fails due to permissions or wrong namespace. [[GH-23503](https://github.com/hashicorp/vault/pull/23503)]
+* website/docs: fix inaccuracies with unauthenticated_in_flight_requests_access parameter [[GH-23287](https://github.com/hashicorp/vault/pull/23287)]
```
Comment (mladlow) [inline CHANGELOG.md:21]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -244,6 +300,48 @@ sdk/ldaputil: use EscapeLDAPValue implementation from cap/ldap [[GH-22249](https
...
+* core: Bump Go version to 1.20.10.
+* replication (enterprise): Switch to non-deprecated gRPC field for resolver target host
+
+IMPROVEMENTS:
+
+* api/plugins: add `tls-server-name` arg for plugin registration [[GH-23549](https://github.com/hashicorp/vault/pull/23549)]
+* core: Use a worker pool for the rollback manager. Add new metrics for the rollback manager to track the queued tasks. [[GH-22567](https://github.com/hashicorp/vault/pull/22567)]
+* core: emit logs when user(s) are locked out and when all lockouts have been cleared [[GH-23140](https://github.com/hashicorp/vault/pull/23140)]
+* ui: Adds toggle to KV secrets engine value download modal to optionally stringify value in downloaded file [[GH-23747](https://github.com/hashicorp/vault/pull/23747)]
+* website/docs: fix inaccuracies with unauthenticated_in_flight_requests_access parameter [[GH-23287](https://github.com/hashicorp/vault/pull/23287)]
```
Comment (mladlow) [inline CHANGELOG.md:317]:
```suggestion
```

**Review by mladlow — APPROVED** [review]
I left some suggestions to remove website updates.

File: CHANGELOG.md [hunk 3]
```diff
@@ -244,6 +299,47 @@ sdk/ldaputil: use EscapeLDAPValue implementation from cap/ldap [[GH-22249](https
...
+CHANGES:
+
+* core: Bump Go version to 1.20.10.
+* replication (enterprise): Switch to non-deprecated gRPC field for resolver target host
+
+IMPROVEMENTS:
+
+* api/plugins: add `tls-server-name` arg for plugin registration [[GH-23549](https://github.com/hashicorp/vault/pull/23549)]
+* core: Use a worker pool for the rollback manager. Add new metrics for the rollback manager to track the queued tasks. [[GH-22567](https://github.com/hashicorp/vault/pull/22567)]
+* core: emit logs when user(s) are locked out and when all lockouts have been cleared [[GH-23140](https://github.com/hashicorp/vault/pull/23140)]
```
Comment (anwittin) [inline CHANGELOG.md:314]: This should have been reverted and it looks like it was not. This can cause a race condition.

Comment (cinthiaconti) [reply CHANGELOG.md:314]: I'm removing this entry from the 1.14.5 changelog since it will be reverted on 1.14.6.

**Pull request merged**